### PR TITLE
Deadlock leader board 294

### PR DESCRIPTION
**Pull request by Todua992**
## Description

We added the Leaderboard Functionality to the app. 
IT was agreed that we ditch the Monthly Ranking System, as Its not something we Go by at the moment
[FIGMA](https://www.figma.com/design/BCEVQYvMDrcBE9qqehr9HL/Mobile-Education-UI?node-id=7713-17908)

## Related Branch PBI
https://github.com/Educado-App/educado-backend/pull/216

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.
- [x] Changes have been approved by Product Owner
- [x] Code has been analyzed with CodeScene
- [x] I have reviewed the pull request myself

## Screenshots (if applicable)

image

## If mobile/frontend pull request, what version of the backend is it stable, and running on? 
**Deadlock-Leaderboard Backend:**
https://github.com/Educado-App/educado-backend/pull/216

## Link to documentation
[MongoDB Agregesion
](https://www.mongodb.com/docs/manual/reference/operator/aggregation/sort/)##

### CODE REVIEW

File: eml/screens/leaderboard/image.png
Comment (LazarShockX) [inline eml/screens/leaderboard/image.png:1]: Very immature

**Review by LazarShockX — APPROVED** [review]
Choose another icon for the left arrow button and align it better. Last part of the name should be replaced with "..." if it exceeds a certain length. I also think names in bold look better to give it a more cartoonish vibe.
Very nicely done 🩵

**Comment by Todua992** [comment]
> Choose another icon for the left arrow button and align it better. Last part of the name should be replaced with "..." if it exceeds a certain length. I also think names in bold look better to give it a more cartoonish vibe. Very nicely done 🩵

I am not allowed to use bold text sadly. Also the names get replaced with ... if they are so long that, they cant even fit in two lines 
Example. I was told, It should be possible to have names on two lines if needed. 

image
image

**Comment by LazarShockX** [comment]
> > Choose another icon for the left arrow button and align it better. Last part of the name should be replaced with "..." if it exceeds a certain length. I also think names in bold look better to give it a more cartoonish vibe. Very nicely done 🩵
> 
> I am not allowed to use bold text sadly. Also the names get replaced with ... if they are so long that, they cant even fit in two lines Example. I was told, It should be possible to have names on two lines if needed.
> 
> image image

Screenshot 2024-11-27 at 18 11 58
Why is this name not replaced with ...?

**Comment by Todua992** [comment]
> > > Choose another icon for the left arrow button and align it better. Last part of the name should be replaced with "..." if it exceeds a certain length. I also think names in bold look better to give it a more cartoonish vibe. Very nicely done 🩵
> > 
> > 
> > I am not allowed to use bold text sadly. Also the names get replaced with ... if they are so long that, they cant even fit in two lines Example. I was told, It should be possible to have names on two lines if needed.
> > image image
> 
> Screenshot 2024-11-27 at 18 11 58 Why is this name not replaced with ...?

Because, It is allowed to be exactly that long, before ... steps in. Thats were we ended up putting.

**Comment by LazarShockX** [comment]
> > > > Choose another icon for the left arrow button and align it better. Last part of the name should be replaced with "..." if it exceeds a certain length. I also think names in bold look better to give it a more cartoonish vibe. Very nicely done 🩵
> > > 
> > > 
> > > I am not allowed to use bold text sadly. Also the names get replaced with ... if they are so long that, they cant even fit in two lines Example. I was told, It should be possible to have names on two lines if needed.
> > > image image
> > 
> > 
> > Screenshot 2024-11-27 at 18 11 58 Why is this name not replaced with ...?
> 
> Because, It is allowed to be exactly that long, before ... steps in. Thats were we ended up putting.

I respect your decision

**Comment by Todua992** [comment]
> > > > > Choose another icon for the left arrow button and align it better. Last part of the name should be replaced with "..." if it exceeds a certain length. I also think names in bold look better to give it a more cartoonish vibe. Very nicely done 🩵
> > > > 
> > > > 
> > > > I am not allowed to use bold text sadly. Also the names get replaced with ... if they are so long that, they cant even fit in two lines Example. I was told, It should be possible to have names on two lines if needed.
> > > > image image
> > > 
> > > 
> > > Screenshot 2024-11-27 at 18 11 58 Why is this name not replaced with ...?
> > 
> > 
> > Because, It is allowed to be exactly that long, before ... steps in. Thats were we ended up putting.
> 
> I respect your decision

No mine decision by ty XD

**Pull request merged**